lib/mix/lib/mix.ex

```diff
@@ -172,21 +172,21 @@ defmodule Mix do
       echo "world!"
 
       # mix.exs
-      defp create_aliases do
+      defp aliases do
         [
-          "taskalias": ["hex.info", "run priv/hello.exs", "cmd priv/world.sh"],
-          "taskalias2": ["run priv/hello1.exs", "run priv/hello2.exs"]
+          "task_alias": ["hex.info", "run priv/hello.exs", "cmd priv/world.sh"],
+          "task_alias2": ["run priv/hello1.exs", "run priv/hello2.exs"]
         ]
       end
 
-  In the example above we have created 2 aliases, the first example
-  `taskalias` will run task `hex.info`, then (`run`)[`Mix.Tasks.Run`]
+  In the example above, we have created two aliases. In the first example,
+  `task_alias` will run task `hex.info`, then (`run`)[`Mix.Tasks.Run`]
   to run an Elixir script, then (`cmd`)[`Mix.Tasks.Cmd`] to run a
-  command line bash script. This shows how powerful aliases mixed
-  with mix tasks can be.
+  command line shell script. This shows how powerful aliases mixed
+  with Mix tasks can be.
 
-  `taskalias2` shows a limitation of tasks where only one of the given
-  tasks will run, the execution of `run priv/hello2.exs` will not run.
```

Comment:
Here it is not clear why this limitation exists

Reply:
All tasks in Mix were design to run only once. So for example, if you have different tasks calling `mix compile`, we won't compile multiple times. This is important so we can do things like `mix test`, that calls a bunch of other tasks that may compile or not, without compiling multiple times.

Reply:
IF you don't want this behaviour, you can always call `Mix.Task.reenable "name_of_your_task"` at the end to reenable it.

Reply:
please have a look at how it looks now

Reply:
Since https://github.com/elixir-lang/elixir/commit/8d082ad3b36392c54d90ef87f638773dff1eb33c, there are some tasks that are automatically reenabled: `run`, `cmd`, `loadconfig`, `xref`, `do`, `profile.cprof`, `profile.eprof`, and `profile.fprof`. The current examples are outdated and invalid, because `task_alias2` would run both `priv/hello1.exs` and `priv/hello2.exs`. I think we should update the examples and mention that some tasks are automatically reenabled.

As a side note, the given example to reenable a task does not work, it raises `** (Mix) The task "Mix.Task.reenable(run)" could not be found`. A working example could be:

```elixir
foo: ["format --check-formatted", "cmd script/generate.sh", fn _ -> Mix.Task.reenable("format") end, "format --check-formatted"]
```

or moving that to a helper function:

```elixir
foo: ["format --check-formatted", "cmd script/generate.sh", &reenable_format_task/1, "format --check-formatted"]

defp reenable_format_task(_) do
  Mix.Task.reenable("format")
end
```

Reply:
@fertapric thank you so much for your detailed suggestion. Have a look at it and tell me what you think.